Fixes attr-for-selected issue and adds animate-first-selected feature.

zerodevx opened the pull request:
Hi @morethanreal,

Wrt issue #8 , here are the proposed code changes for your reference.

Thanks so much,
Jason

morethanreal commented on neon-animated-pages.html:
```diff
@@ -81,54 +89,64 @@
-        this.playAnimation();
+        //this.playAnimation();
+        // in the event of first load, we want to run animations only after current microtasks are complete (ie element attached)
+        this.async(this.playAnimation);
```
Can it only async if one of the pages is not attached? Check for `isAttached` maybe?

morethanreal commented on neon-animated-pages.html:
```diff
@@ -55,7 +55,7 @@
 
     is: 'neon-animated-pages',
-
+    
```
Can you remove white spaces from line endings?

morethanreal commented on neon-animated-pages.html:
```diff
@@ -67,11 +67,19 @@
+       * @type boolean
+       * @default false
+       */
+      animateFirstSelected: {
```
Maybe rename to `animate-initial-selection`?

morethanreal commented on neon-animated-pages.html:
```diff
@@ -67,11 +67,19 @@
+       *
+       * @attribute animate-first-selected
+       * @type boolean
+       * @default false
```
I think hydrolysis is able to parse the property block without these annotations.

morethanreal commented:
Thanks for your contribution. In addition to the comments, would you mind splitting the animate first selection feature and the fix for attr-for-selected into separate commits? I also committed a fix for https://github.com/PolymerElements/neon-animation/issues/11, so you'll need to rebase. Thank you!

zerodevx commented:
Thanks for your notes, @morethanreal. I've

1. Renamed feature to `animate-first-selection`;
2. Split into [fix-for-attr-selected] (https://github.com/zerodevx/neon-animation/tree/fix-attr-for-selected) and [add-animate-initial-selection] (https://github.com/zerodevx/neon-animation/tree/add-animate-initial-selection) branches.
3. Removed white-spaces;
4. Added an `isAttached` check so that async is run only if the element is not yet attached;
5. Removed annotations.

I'll submit the PR for `animate-initial-selection` in a bit.

Thanks so much,
Jason

morethanreal commented:
lgtm
